### DISCUSSION TITLE
[mobx-undecorate] Enable mobx-undecorate to Run from Anywhere in Project

### DISCUSSION THREAD

**Discussion by ITenthusiasm** [mobxjs/mobx#2884] [Ideas]
It wasn't readily apparent where to find the file for the code change, so I'm starting out with an issue that has a code change suggestion.

Right now, `mobx-undecorate` doesn't seem to work if the user isn't in the correct folder. The `cli.js` script should be updated to enable the user to run the script from anywhere in the project. (This will make it easier to use and prevent potential confusion, as many packages work as long as you're somewhere in a node project.)

```diff
#!/usr/bin/env node
const { spawn } = require("child_process")
const path = require("path")

spawn(
-    path.join("node_modules", ".bin", "jscodeshift"),
+    `"${path.resolve(__dirname, "../.bin/jscodeshift")}"`,
    [
        "--extensions=js,jsx,ts,tsx",
        ...process.argv.filter(arg => arg.startsWith("--")),
        "-t",
        "src/undecorate.ts",
        process.cwd()
    ],
    {
        cwd: __dirname,
        stdio: "inherit",
        shell: true
    }
)
```

The implementation above should enable people to run the script from anywhere in the project. Quotation marks are included to enable users who have folders with spaces along the path (eg. `My Dangerous Folder Name`) to run the script.

Happy to submit a PR if I can get direction to the right location. (Also happy to take suggestions if for some reason this seems like an undesirable idea.)

**Comment by ITenthusiasm** [comment]
I still haven't gotten used to how repos with `yarn` work yet. 🤦🏾‍♂️ I see where the file is. I'll submit a PR myself.

**Comment by urugator** [comment]
I think #2818 handles that. It will be part of the next release #2842

**Reply by mweststrate** [reply]
Should be released in < 1 hour

**Reply by ITenthusiasm** [reply]
Okay cool 👍🏾
